cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -420,6 +419,260 @@ class self_comparator {
 };
 
 }  // namespace lexicographic
+
+namespace equality_hashing {
```

Comment:
Can we just have separate `equality` and `hash` namespaces? 

The same `preprocessed_table` type can be used in both namespaces by injecting it from one namespace to the other. 

```
namespace equality{
   struct preprocess_table{ /* definition here */ };
}

namespace hash{
   using equality::preprocessed_table;
}

// These are the same type
hash::preprocessed_table t;
equality::preprocessed_table;
```

This way if for any reason the `hash` vs `equality` preprocessed tables need to be different in the future, it is easier to make that change.